### PR TITLE
feat: upgrade express, dd-trace, body-parser

### PR DESCRIPTION
**Pull request by bodinsamuel**
## Changes

- Upgrade express to v5
It's mostly similar but
  - can't reasign req.params, req.query, req.body. It was an issue for `validateRequest`, so I had to put everything in res.locals (which is more correct anyway)
  - can't put `never` in Locals anymore (not sure why)
  - `/*` -> `/*splat` wildcard must be named and can't be accessed like before
 
- Upgrade dd-trace
Needed for compatibly 

- Upgrade body-parser
Needed for compatibly 

## 🧪 Tests

Please do test locally
- UI navigate everywhere and send form
- API trigger action, sync, proxy and action endpoint
- If you have, try to upload a file to an integration

---

**Major Upgrade: Express v5, dd-trace, and body-parser with Refactor to res.locals Across Backend**

This PR introduces a sweeping upgrade across the Nango monorepo, migrating all major backend services (orchestrator, jobs, persist, runner, keystore, fleet) from Express v4 to v5, along with dd-trace and body-parser version updates. The transition required extensive refactoring: all middleware, route handlers, and validation utilities now extract and validate HTTP input (params, query, body) via res.locals (using names like parsedBody, parsedParams, etc.) instead of mutating or directly referring to req.body, req.params, or req.query—a pattern now prohibited by Express v5. Additionally, wildcard routes are updated per Express 5 conventions, and all relevant package.json files have synchronized dependency updates. Automated and manual testing of all endpoints is strongly encouraged given the scope.

**Key Changes:**
• Upgrade all backend dependencies to Express 5.1.0, dd-trace 5.52.0, and latest body-parser where applicable
• Systematic refactor of all ``HTTP`` route handlers and middleware to use res.locals for validated input; direct mutation/access of req.body, req.params, req.query eliminated
• Validation logic (zod, etc.) moved or updated for compatibility and stricter type safety
• All wild-card routes rewritten to Express 5 syntax (e.g., /* → /*splat)
• Middleware and handler signatures updated to match new Express 5 requirements
• Package.json dependencies upgraded and aligned across orchestrator, jobs, persist, runner, keystore, and fleet packages
• Improved import hygiene and module orderings throughout

**Affected Areas:**
• orchestrator, jobs, persist, runner, keystore, fleet ``HTTP`` route files, and associated middlewares
• Custom auth, logging, and validation middlewares
• Route validation utilities (zod schemas and their usage)
• All Express input handling code
• Dependency and build configuration (package.jsons)

**Potential Impact:**

**Functionality**: High; any missed migration to res.locals or missed handler/middleware update may lead to request input missing or being unvalidated. All API endpoint logic affected.

**Performance**: No major regressions expected; slight improvements possible due to newer Express internals.

**Security**: Potential improvement as all input is funneled through explicit validation, but any incomplete migration may cause validation bypass or unhandled errors.

**Scalability**: No direct changes, but all deployment environments must reinstall dependencies and retest; potential impacts if deployment scripts assume older Express.

**Review Focus:**
• Thorough adoption of res.locals for all body, params, and query handling
• No residual req.body/req.params/req.query mutation or read before validation
• Wild-card route patterns updated correctly and covered in tests
• Middleware order and type safety (esp. in large handler/middleware chains)
• Dependency and lockfile consistency, esp. in package.json updates

<details>
<summary><strong>Testing Needed</strong></summary>

• Full regression suite for all ``API`` endpoints and workflows in each backend service
• Manual ``UI`` navigation, form posting, and file upload (all noted in ``PR``)
• Manual and automated integration tests to cover custom and edge-case middleware flows
• Wild-card and custom parameter route handling should be checked carefully
• Middleware ordering (e.g., ``JSON`` parsing, auth) must be validated

</details>

<details>
<summary><strong>Code Quality Assessment</strong></summary>

**packages/jobs/lib/routes/tasks/putTask.ts**: All critical validation and input type enforcement now use res.locals; strict checking for required fields and error management improved.

**packages/orchestrator/lib/routes/v1/postImmediate.ts**: All body input moved to res.locals.parsedBody; handler now adheres to stricter input safety and validation, imports reordered for clarity.

**packages/orchestrator/lib/routes/v1/tasks/taskId/getOutput.ts**: Successfully migrated to res.locals for long polling, no direct param access, and better validation integration; handler signature and event logic adapted as required.

**packages/persist/lib/server.ts**: Refactored for clarity, all route and middleware signatures and mount points follow new Express conventions, wildcards are modernized, and import hygiene improved.

</details>

<details>
<summary><strong>Best Practices</strong></summary>

**Middleware**:
• Order of auth and body parsing is explicit and enforced

**Validation**:
• zod schemas consistently applied before usage

**Type Safety**:
• All route handlers explicitly typed, `TypeScript` types enforced

**Dependency Management**:
• Version pinning and lockstep upgrades across packages

**Imports**:
• Ordering, deduplication, and hygiene improved throughout

</details>

<details>
<summary><strong>Possible Issues</strong></summary>

• Any missed refactor to res.locals may break routes or expose unvalidated input
• Unmigrated custom middleware relying on Express 4 internals may fail
• Wildcard route changes could cause routing failures if any were overlooked
• Ergonomics: res.locals is more verbose; API is less convenient, as noted in review comments
• Naming conventions (parsedBody, parsedParams, ...) should be consistent

</details>

---
*This summary was automatically generated by @propel-code-bot*

### CODE REVIEW

File: packages/utils/lib/express/route.ts [hunk 1]
```diff
@@ -1,23 +1,28 @@
...
 
-export type EndpointRequest<E extends Endpoint<any>> = Request<E['Params'], E['Reply'], E['Body'], E['Querystring']>;
-export type EndpointResponse<E extends Endpoint<any>, Locals extends Record<string, any> = Record<string, never>> = Response<E['Reply'], Locals>;
+import type { Endpoint } from '@nangohq/types';
+import type { Express, NextFunction, Request, Response } from 'express';
+
+export type EndpointRequest = Request<never, never, never, never>;
+export type EndpointResponse<E extends Endpoint<any>, Locals extends Record<string, any> = Record<string, never>> = Response<
+    E['Reply'],
+    Locals & { body: E['Body']; params: E['Params']; query: E['Querystring'] }
```
Comment (TBonnin) [inline packages/utils/lib/express/route.ts]: would `parsedBody`, `parsedParams`, ... be better names?

File: packages/utils/lib/express/route.ts [hunk 2]
```diff
@@ -1,23 +1,28 @@
...
 export interface Route<E extends Endpoint<any>> {
     path: E['Path'];
     method: E['Method'];
 }
 
 export interface RouteHandler<E extends Endpoint<any>, Locals extends Record<string, any> = Record<string, never>> extends Route<E> {
-    validate: (req: EndpointRequest<E>, res: EndpointResponse<E, Locals>, next: NextFunction) => void;
-    handler: (req: EndpointRequest<E>, res: EndpointResponse<E, Locals>, next: NextFunction) => void | Promise<void>;
+    validate: (req: EndpointRequest, res: EndpointResponse<E, Locals>, next: NextFunction) => void;
+    handler: (req: EndpointRequest, res: EndpointResponse<E, Locals>, next: NextFunction) => void | Promise<void>;
```
Comment (TBonnin) [inline packages/utils/lib/express/route.ts:26]: looks like `req` is not used anymore. Should we just remove it or keep it just in case?

File: packages/jobs/lib/routes/runners/postIdle.ts [hunk 3]
```diff
@@ -1,16 +1,19 @@
...
 
 const validate = validateRequest<PostIdle>({
     parseParams: (data) => z.object({ nodeId: z.coerce.number().positive() }).strict().parse(data)
 });
 
-const handler = async (req: EndpointRequest<PostIdle>, res: EndpointResponse<PostIdle>) => {
+const handler = async (_req: EndpointRequest, res: EndpointResponse<PostIdle>) => {
     try {
-        const idle = await runnersFleet.idleNode({ nodeId: req.params.nodeId });
+        const idle = await runnersFleet.idleNode({ nodeId: res.locals.params.nodeId });
```
Comment (TBonnin) [inline packages/jobs/lib/routes/runners/postIdle.ts]: not as nice of an API  unfortunately

**Review by TBonnin — APPROVED** [review]
> can't reasign req.params, req.query, req.body

Isn't it only req.query that's a getter now?
There also seem to be an ugly workaround with 
```
Object.defineProperty(req, 'query', { ...Object.getOwnPropertyDescriptor(req, 'query'), value: req.query, writable: true });
```
that could be applied when query must be parsed

File: packages/utils/lib/express/route.ts (hunk 2 above)
Comment (bodinsamuel) [reply packages/utils/lib/express/route.ts:26]: I think keeping it is safer, otherwise it will be annoying when we'll need it again next week :D

File: packages/jobs/lib/routes/runners/postIdle.ts (hunk 3 above)
Comment (bodinsamuel) [reply packages/jobs/lib/routes/runners/postIdle.ts]: yeah, that's the game tbh

**Comment by bodinsamuel** [comment]
> Isn't it only req.query that's a getter now?

Yeah but no reason for us to continue doing that for the others. Modifying `req` was never recommended anyway

> There also seem to be an ugly workaround with

🫣
